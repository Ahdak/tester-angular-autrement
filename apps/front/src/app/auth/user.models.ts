export interface AuthenticateBody {
  username: string;
  password: string;
}

export interface AuthenticateResponse {
  accessToken: string;
  user: UserModel;
}

export interface UserModel {
  username: string;
  firstname: string;
  lastname: string;
  isPremium: boolean;
}
