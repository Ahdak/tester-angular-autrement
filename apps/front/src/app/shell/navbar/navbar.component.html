<mat-toolbar color="primary" class="mat-elevation-z6">
  <a routerLink="/" mat-button class="company">
    <mat-icon>pool</mat-icon> <span data-testid="company-name">WaterBnb</span>
  </a>
  <span class="catchword">Locations les pieds dans l'eau</span>
  <span class="fill-remaining-space"></span>
  <a *ngIf="!(loggedIn$ | async); else logoutTpl" routerLink="/login" mat-button data-testid="login">
    <mat-icon>account_circle</mat-icon>
  </a>
  <ng-template #logoutTpl>
    <ng-container *ngIf="user$ | async as user">
      <i *ngIf="user.isPremium" class="material-icons premium" data-testid="premium">stars</i>
      <span data-testid="user">{{user.firstname}} {{user.lastname}}</span>
    </ng-container>
    <button (click)="logoutUser()" mat-mini-fab class="logout" data-testid="logout">
      <mat-icon aria-label="Logout">power_off</mat-icon>
    </button>
  </ng-template>
</mat-toolbar>

