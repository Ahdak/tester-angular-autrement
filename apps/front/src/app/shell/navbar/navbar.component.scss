.company {
  font-weight: 700;
  font-size: 17px;
}

.fill-remaining-space {
  flex: 1 1 auto;
}

.catchword {
  font-size: 14px;
  text-shadow: 4px 4px 4px #aaa;
}

.premium {
  color: yellow;
  margin: 2px 5px 0px 0px;
}

.logout {
  margin-left: 20px;
}
