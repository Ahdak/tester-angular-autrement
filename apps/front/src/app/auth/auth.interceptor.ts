import { Injectable } from '@angular/core';
import {
  HttpInterceptor,
  HttpRequest,
  HttpHandler,
} from '@angular/common/http';
import { mergeMap, take } from 'rxjs/operators';

import { AuthFacade } from './+state/auth.facade';

@Injectable()
export class AuthInterceptor implements HttpInterceptor {
  static readonly tokenHeaderName = 'x-auth-token';

  constructor(private authFacade: AuthFacade) {}

  intercept(req: HttpRequest<any>, next: HttpHandler) {
    return this.authFacade.accessToken.pipe(
      take(1),
      mergeMap(accessToken => {
        const authReq = req.clone(
          accessToken
            ? {
                headers: req.headers.set(
                  AuthInterceptor.tokenHeaderName,
                  accessToken,
                ),
              }
            : {},
        );
        return next.handle(authReq);
      }),
    );
  }
}
