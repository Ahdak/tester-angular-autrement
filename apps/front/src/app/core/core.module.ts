import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HttpClientModule } from '@angular/common/http';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { NgProgressModule } from '@ngx-progressbar/core';
import { NgProgressRouterModule } from '@ngx-progressbar/router';
import { SharedModule } from '../shared/shared.module';
import { AppStoreModule } from './store/store.module';
import { AuthModule } from '../auth/auth.module';

@NgModule({
  imports: [
    CommonModule,
    BrowserModule,
    BrowserAnimationsModule,
    HttpClientModule,
    NgProgressModule.forRoot(),
    NgProgressRouterModule,
    AppStoreModule,
    AuthModule,
    SharedModule,
  ],
})
export class CoreModule {}
