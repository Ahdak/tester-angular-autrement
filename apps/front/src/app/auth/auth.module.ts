import { NgModule } from '@angular/core';
import { SharedModule } from '../shared/shared.module';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';
import { authReducers } from './+state/auth.selectors';
import { AuthEffects } from './+state/auth.effects';
import { LoginFormComponent } from './login-form/login-form.component';
import { LoginPageComponent } from './login-page/login-page.component';

@NgModule({
  imports: [
    SharedModule,
    StoreModule.forFeature('auth', authReducers),
    EffectsModule.forFeature([AuthEffects]),
  ],
  declarations: [LoginFormComponent, LoginPageComponent],
})
export class AuthModule {}
