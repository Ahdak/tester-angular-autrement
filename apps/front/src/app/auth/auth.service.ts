import { Injectable } from '@angular/core';
import { Observable, of } from 'rxjs';
import { HttpClient } from '@angular/common/http';
import { AuthenticateBody, AuthenticateResponse } from './user.models';

@Injectable({
  providedIn: 'root',
})
export class AuthService {
  constructor(private httpClient: HttpClient) {}

  login(body: AuthenticateBody): Observable<AuthenticateResponse> {
    return this.httpClient.post<AuthenticateResponse>(
      '/api/authenticate',
      body,
    );
  }

  logout() {
    return of(true);
  }
}
