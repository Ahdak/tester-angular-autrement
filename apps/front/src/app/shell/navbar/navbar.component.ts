import { Component } from '@angular/core';
import { AuthFacade } from '../../auth/+state/auth.facade';

@Component({
  selector: 'app-navbar',
  templateUrl: './navbar.component.html',
  styleUrls: ['./navbar.component.scss'],
})
export class NavbarComponent {
  loggedIn$ = this.authFacade.loggedIn$;
  user$ = this.authFacade.user$;

  constructor(private authFacade: AuthFacade) {}

  logoutUser() {
    this.authFacade.logout();
  }
}
