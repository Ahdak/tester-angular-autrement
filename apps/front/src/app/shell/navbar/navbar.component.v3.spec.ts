import { NO_ERRORS_SCHEMA } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NavbarComponent } from './navbar.component';
import { AuthFacade } from '../../auth/+state/auth.facade';
import { BehaviorSubject } from 'rxjs';
import { UserModel } from '../../auth/user.models';

describe('NavbarComponent Snapshot testing way', () => {
  let component: NavbarComponent;
  let fixture: ComponentFixture<NavbarComponent>;
  let authFacade: AuthFacade;
  let nativeElement: Element;

  const loggedIn$ = new BehaviorSubject<boolean>(false);
  const user$= new BehaviorSubject<UserModel>(null);

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [NavbarComponent],
      providers: [
        { provide: AuthFacade, useValue: { loggedIn$, user$, logout: jest.fn() }},
      ],
      schemas: [NO_ERRORS_SCHEMA]
    }).compileComponents();

    fixture = TestBed.createComponent(NavbarComponent);
    component = fixture.componentInstance;
    nativeElement = fixture.nativeElement;
    authFacade = TestBed.get(AuthFacade);
  }));

  it('should create match initail snapshot', () => {
    fixture.detectChanges();
    expect(nativeElement).toMatchSnapshot();
  });

  it('should match snapshot when logged in', () => {
    loggedIn$.next(true);
    user$.next({username: 'mbob', firstname: 'Bob', lastname: 'Marley', isPremium: false});
    fixture.detectChanges();
    expect(nativeElement).toMatchSnapshot();
  });

});
