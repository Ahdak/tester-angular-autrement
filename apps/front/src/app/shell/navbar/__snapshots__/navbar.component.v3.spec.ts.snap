// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarComponent Snapshot testing way should create match initail snapshot 1`] = `
<div
  id="root0"
  ng-version="6.1.10"
>
  <mat-toolbar
    class="mat-elevation-z6"
    color="primary"
  >
    <a
      class="company"
      mat-button=""
      routerlink="/"
    >
      <mat-icon>
        pool
      </mat-icon>
      <span
        data-testid="company-name"
      >
        WaterBnb
      </span>
    </a>
    <span
      class="catchword"
    >
      Locations les pieds dans l'eau
    </span>
    <span
      class="fill-remaining-space"
    />
    <!--bindings={
  "ng-reflect-ng-if": "true",
  "ng-reflect-ng-if-else": "[object Object]"
}-->
    <a
      data-testid="login"
      mat-button=""
      routerlink="/login"
    >
      <mat-icon>
        account_circle
      </mat-icon>
    </a>
    <!---->
  </mat-toolbar>
</div>
`;

exports[`NavbarComponent Snapshot testing way should match snapshot when logged in 1`] = `
<div
  id="root1"
  ng-version="6.1.10"
>
  <mat-toolbar
    class="mat-elevation-z6"
    color="primary"
  >
    <a
      class="company"
      mat-button=""
      routerlink="/"
    >
      <mat-icon>
        pool
      </mat-icon>
      <span
        data-testid="company-name"
      >
        WaterBnb
      </span>
    </a>
    <span
      class="catchword"
    >
      Locations les pieds dans l'eau
    </span>
    <span
      class="fill-remaining-space"
    />
    <!--bindings={
  "ng-reflect-ng-if": "false",
  "ng-reflect-ng-if-else": "[object Object]"
}-->
    <!--bindings={
  "ng-reflect-ng-if": "[object Object]"
}-->
    <!---->
    <!--bindings={
  "ng-reflect-ng-if": "false"
}-->
    <span
      data-testid="user"
    >
      Bob Marley
    </span>
    <button
      class="logout"
      data-testid="logout"
      mat-mini-fab=""
    >
      <mat-icon
        aria-label="Logout"
      >
        power_off
      </mat-icon>
    </button>
    <!---->
  </mat-toolbar>
</div>
`;
